gmusicapi/clients/mobileclient.py

```diff
@@ -123,7 +130,7 @@ def login(self, email, password, android_id, locale='en_US'):
             self.logger.info("failed to authenticate")
             return False
 
-        self.android_id = android_id
+        self.android_id = self._validate_device_id(android_id)
```

Comment:
I think raising an exception here may put us into an odd state where the underlying session is authed but the client isn't.

I was thinking at first to just move this check up, but we need to be authed to actually make it. Maybe we catch the exception here, then log out and reraise if it happens?